allure-generator/src/main/javascript/data/tree/TreeCollection.js

```diff
@@ -69,4 +74,54 @@ export default class TreeCollection extends Collection {
             return this.testcases[index - 1];
         }
     }
+
+    calculateStatistic(items) {
+        const statistic = {};
+        values.forEach(value => {
+            statistic[value] = 0;
+        });
+        items.forEach(item => {
+            if (item.children) {
+                values.forEach(value => {
+                    statistic[value] += item.statistic ? item.statistic[value] : 0;
+
+                });
+            } else {
+                statistic[item.status]++;
+            }
+        });
+        return statistic;
+    }
+
+    calculateTime(items) {
+        const time = {
+            maxDuration: Number.MIN_VALUE,
+            minDuration: Number.MAX_VALUE,
+            sumDuration: 0,
+            duration: 0,
+            start: Number.MAX_VALUE,
+            stop: Number.MIN_VALUE
+        };
+        items.forEach(item => {
+            if (item.children) {
+                TreeCollection.updateTime(time, item.time, 'maxDuration', Math.max);
+                TreeCollection.updateTime(time, item.time, 'minDuration', Math.min);
+                TreeCollection.updateTime(time, item.time, 'sumDuration', (a, b) => a + b);
+            } else if (item.time && item.time.duration) {
+                time.maxDuration = Math.max(time.maxDuration, item.time.duration);
+                time.minDuration = Math.min(time.minDuration, item.time.duration);
+                time.sumDuration = time.sumDuration + item.time.duration;
+            }
+            TreeCollection.updateTime(time, item.time, 'start', Math.min);
+            TreeCollection.updateTime(time, item.time, 'stop', Math.max);
+            time.duration = time.stop - time.start;
+        });
+        return time;
+    }
+
+    static updateTime(timeA, timeB, field, operation) {
```

Comment:
This is not Java you don't need to use static methods, you can just create a standalone function within this module